src/ucp/dt/dt_iov.c

```diff
@@ -36,3 +37,31 @@ void ucp_dt_iov_memcpy(void *dest, const ucp_dt_iov_t *iov, size_t length,
         }
     }
 }
+
+size_t ucp_dt_iov_scatter(ucp_dt_iov_t *iov, size_t iovcnt, const void *src,
+                          size_t length, size_t *iov_offset, size_t *iovcnt_offset)
+{
+    size_t item_len, item_len_to_copy;
+    size_t length_it = 0;
+
+    while ((length_it < length) && (*iovcnt_offset < iovcnt)) {
+        item_len         = iov[*iovcnt_offset].length;
+        item_len_to_copy = item_len ? ucs_min(item_len - *iov_offset, length -
```

Comment:
need to cast the arguments to ssize_t, this way could also avoid checking if item_len==0